tools/ci/generateCppPackagesConfig.ps1

```diff
@@ -0,0 +1,57 @@
+# Generate a custom packages.config to restore only the necessary Core
+# NuGet packages for the given build triple, to save on disk space
+
+param(
+    [ValidateSet('Win32','UWP')]
+    [string]$BuildPlatform,
+    [ValidateSet('x86','x64','ARM')]
+    [string]$BuildArch,
+    [ValidateSet('Debug','Release')]
+    [string]$BuildConfig,
+    [string]$InputFile,
+    [string]$OutputFile
+)
+
+Write-Host "Generating packages.config for build triple $BuildPlatform-$BuildArch-$BuildConfig..."
+
+$reader = [System.IO.File]::OpenText($InputFile)
+$content = ""
+try {
+    for() {
+        $line = $reader.ReadLine()
+        if ($line -eq $null) { break }
+        if ($line -match "Microsoft\.MixedReality\.WebRTC\.Native\.Core\.(Desktop|UWP|WinRT)\.")
+        {
+            if ($line -match "Microsoft\.MixedReality\.WebRTC\.Native\.Core\.(Desktop|UWP|WinRT)\.$BuildArch\.$BuildConfig")
```

Comment:
Aren't these redundant?

Reply:
Oh I see what it does. I'd rather do something like
```
if ($line -match "Microsoft\.MixedReality\.WebRTC\.Native\.Core\.(Desktop|UWP|WinRT)\.(.*)\.(.*)")
  if ($Matches.1 != ... OR $Matches.2 != $BuildArch OR $Matches.3 != $BuildConfig)
    // reject
// accept

Reply:
This is slightly more subtle than that, because for all UWP variants we need to restore the package `Microsoft.MixedReality.WebRTC.Native.Core.UWP` which contains the UWP generated headers (and no lib). With your proposed change, the line for this package will match, but `$Matches.2` will be `1` and `$Matches.3` will be `0`, because the full packages path is `Microsoft.MixedReality.WebRTC.Native.Core.UWP.1.0.3`. So the first `-match` here is to exclude that package, which is treated as any other ignored package and copied as is. Then for packages which contain `webrtc.lib` or `Org.WebRtc.dll` and are per-config/per-arch, we do the second match.

Reply:
All of this would look good in a comment 😄

Reply:
Fixed.